memq/src/main/java/com/pinterest/memq/core/clustering/Balancer.java

```diff
@@ -55,6 +55,8 @@ public Balancer(MemqConfig config,
     this.readBalanceStrategy = config.getClusteringConfig().isEnableExpiration()
         ? new ExpirationPartitionBalanceStrategyWithAssignmentFreeze()
         : new PartitionBalanceStrategy();
+    this.writeBalanceStrategy.setMemqConfig(config);
```

Comment:
is this the only place that `new Balancer()` constructors are called? if so, I would suggest passing the config into the constructor, and making the memqConfig variable `final`. if not, we should do the null check as I mentioned in the other comment.

Reply:
i assume you are referring to BalanceStrategy. Yes, this is the only place that it's called. I didn't plan to update the default constructor because these values are not mandatory for the strategy logic, plus the BalanceStrategy classes do not have constructor in code... 

I agree this value should be final. I will add two constructors.